### PR TITLE
wishlist functionality

### PR DESCRIPTION
**Pull request by arpit01923**
Added Functionality-

1. Now you should remove and add item from product page 
2. And remove item from from wishlist page
3. if there is no item, then it shows wishlist is empty
4. here the counter of navbar should also affect

Preview-

https://user-images.githubusercontent.com/86866251/161381046-7cd8f68f-3d89-445b-8917-657b95c26be9.mp4

### CODE REVIEW

File: src/pages/authentication/Logout.jsx [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 
-        <a className="secondary" href="#">
-          {" "}
-          Log Me Out{" "}
-        </a>
-      </form>
-    </section>
+      <a className="secondary" href="#">
+        {" "}
+        Log Me Out{" "}
```
Comment (hrishikeshmane) [inline src/pages/authentication/Logout.jsx]: remove empty spaces if not required

File: src/reducer/WishlistReducer.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {addToWishlist,removeFromWishlist} from "utils/wishlist";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "Add_to_wishlist":
```
Comment (hrishikeshmane) [inline src/reducer/WishlistReducer.jsx]: Right convention would be- "ADD_TO_WISHLIST"

File: src/utils/wishlist.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const addToWishlist = (wishlist, payload) => {
+  return wishlist.find((product) => product._id === payload._id)
+    ? wishlist
+    : [...wishlist, payload];
+}
+
+const removeFromWishlist = (wishlist, payload) => {
+  console.log({ wishlist, payload });
```
Comment (hrishikeshmane) [inline src/utils/wishlist.jsx]: remove console logs

**Review by hrishikeshmane — COMMENTED** [review]
Use prettier to format code

File: src/reducer/WishlistReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import {addToWishlist,removeFromWishlist} from "utils/wishlist";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case  "ADD_TO_WISHLIST":
+      return {
+        ...state,
+        wishlist: addToWishlist(state.wishlist, action.payload),
```
Comment (nyyshaaa) [inline src/reducer/WishlistReducer.jsx:8]: i guess you should use ... before wishlist state to make sure it's being cloned while passing arguments.
Not sure though, you can confirm.

File: src/pages/wishlist/main/Card.jsx [hunk 5]
```diff
@@ -1,42 +1,50 @@
...
-    <main class="sub-container">
-      {wishlists.map((wishlist) => (
-        <>
-          <section class="padding-0 card horizontal">
-            <img class="img" src={wishlist} alt="wishlist-products" />
+const Card = () => {
+  const { state, dispatch } = useWishlist();
+  return (
+    <main className="sub-container">
+      {console.log({ state })}
```
Comment (nyyshaaa) [inline src/pages/wishlist/main/Card.jsx]: remove console

File: src/reducer/WishlistReducer.jsx (hunk 4 above)
Comment (arpit01923) [reply src/reducer/WishlistReducer.jsx:8]: not needed

**Pull request merged**